packages/chain-adapters/src/types.ts

```diff
@@ -16,6 +16,10 @@ import * as utxo from './utxo/types'
 
 export { cosmossdk, evm, utxo }
 
+// this placeholder forces us to be explicit about transactions not transferring funds to humans
+export type ContractInteraction = Nominal<'contract-interaction', 'ContractInteraction'>
+export const CONTRACT_INTERACTION: ContractInteraction = 'contract-interaction' as const
```

Comment:
i wonder if we should be simulating the tx to gather addresses and screen those than relying on us to manually specify a single receive address? a follow-up perhaps?

Reply:
Would definitely make sense for UTXOs (i.e "simulating" as in, coinselect algorithm will give us the input/output addresses).
For EVMs, since we effectively know the receiving address beforehand, I don't believe this is needed.
For Cosmos SDK chains, simulating is impossible and would be effectively useless, since there are no contracts per se, and the only reason we know something is a "Contractish" is because we know that we're dealing with e.g a `MsgDeposit`.

Reply:
Although one nuance here for UTXOs, same thing, there's no contract per se, but if we're dealing with e.g THOR savers deposit, then this is effectively a "Contractish" call, and we know it since we're the ones building said Tx